### DISCUSSION TITLE
unfortunate bug in KDE that incorrectly makes Ghostty look like it is infinite looping

### DISCUSSION THREAD

**Discussion by andrewrk** [ghostty-org/ghostty#2932] [Feature Requests, Ideas] [answered]
I upgraded to Plasma 6.2 today and unfortunately still seeing this:

![image](https://github.com/user-attachments/assets/de67f836-b1f2-4906-a279-2874066b94b7)

It looks like 2 threads infinite looping per ghostty window.

It's certainly a bug in the Individual Core Usage plugin because it does not show any ghostty CPU usage in the system monitor application, or in htop.

I found [this discussion](https://discuss.kde.org/t/individual-core-usage-doesnt-match-top-by-a-lot/9232) but no bug report.

I was kinda hoping since KDE is popular that I wouldn't have to do any work myself to fix this but it's starting to look like I'm the only person who cares about this enough to do something about it.

Anyway just starting this thread to see if anyone else is running into this.

**Comment by andrewrk** [comment]
I [reported the bug](https://bugs.kde.org/show_bug.cgi?id=497341) but it was a little bit naughty to do it before the public release of Ghostty.

**Reply by ssergiienko** [reply]
> It looks like 2 threads infinite looping per ghostty window.

@andrewrk Seems like a ghostty bug, not KDE (or I misunderstood your issue). Take a look at https://github.com/ghostty-org/ghostty/discussions/3224 and https://github.com/ghostty-org/ghostty/discussions/3658

**Comment by henbr** [comment]
I've the same issue, KDE is showing very high overall CPU usage while at the same time the ghostty process is at 0%.

A workaround that seems to "fix" it is to compile ghostty/libxev to use epoll instead of io_uring as backend.

**Reply by ssergiienko** [reply]
> to "fix" it is to compile ghostty/libxev to use epoll instead of io_uring as backend.

For some reason recompiling [didn't fix](https://github.com/ghostty-org/ghostty/discussions/3224#discussioncomment-11684863) the issue for me.

**UPD:** it works, just required extra shell option for glob

**Comment by skytwosea** [comment]
Chiming in to report that I am seeing this same issue: see discussion #4337 for a full breakdown and system/environment description

**Comment by pluiedev** [comment, marked as answer]
Closing in favor of #3224
